logger.js

```diff
@@ -58,10 +62,17 @@ function pinoLogger (opts, stream) {
     res.startTime = Date.now()
     if (!req.res) { req.res = res }
 
-    res.on('finish', onResFinished)
-    res.on('error', onResFinished)
-    // it's possible that browser aborts connection, or http-server because of timeout
-    req.on('aborted', onReqAborted)
+    res.once('finish', onResFinished)
+    res.once('error', onResFinished)
+
+    req.once('close', onReqClosed)
+    req.once('aborted', onReqAborted)
+    req.onSocketTimeout = function () {
+      req.removeListener('aborted', onReqAborted)
+      res.statusCode = 408
+      onResFinished.call(res, new Error('Timeout'))
+    }
+    req.socket.once('timeout', req.onSocketTimeout)
```

Comment:
Node uses `socket.on('timeout', ...)` to implement `res.on('timeout')` as well as `req.on('timeout')`. https://github.com/nodejs/node/blob/a4bb9fdb893c8f2c36a6b77862e29b2608e080c2/lib/_http_server.js#L285